debugging/kernel-logger/kp_kernel_logger.cpp

```diff
@@ -19,14 +19,82 @@
 #include <vector>
 #include <string>
 #include <limits>
+#include <climits>
 #include <cstring>
+#include "impl/Kokkos_Profiling_Interface.hpp"
 
 std::vector<std::string> regions;
 static uint64_t uniqID;
 struct SpaceHandle {
   char name[64];
 };
 
+// Get a useful label from the deviceId
+// NOTE: Relevant code is in:
+// kokkos/core/src/impl/Kokkos_Profiling_Interface.hpp
+std::string deviceIdToString(const uint32_t deviceId) {
+  using namespace Kokkos::Tools::Experimental;
+  std::string device_label("(");
+  ExecutionSpaceIdentifier eid = identifier_from_devid(deviceId);
+  if (eid.type == DeviceType::Serial)
+    device_label += "Serial";
+  else if (eid.type == DeviceType::OpenMP)
+    device_label += "OpenMP";
+  else if (eid.type == DeviceType::Cuda)
+    device_label += "Cuda";
+  else if (eid.type == DeviceType::HIP)
+    device_label += "HIP";
+  else if (eid.type == DeviceType::OpenMPTarget)
+    device_label += "OpenMPTarget";
+  else if (eid.type == DeviceType::HPX)
+    device_label += "HPX";
+  else if (eid.type == DeviceType::Threads)
+    device_label += "Threads";
+  else if (eid.type == DeviceType::SYCL)
+    device_label += "SYCL";
+  else if (eid.type == DeviceType::OpenACC)
+    device_label += "OpenACC";
+  else if (eid.type == DeviceType::Unknown)
+    device_label += "Unknown";
+  else
+    device_label += "Unknown to KokkosTools";
+
+  if (eid.instance_id ==
+      int_for_synchronization_reason(
+          SpecialSynchronizationCases::GlobalDeviceSynchronization))
+    device_label += " All Instances)";
+  else if (eid.instance_id ==
+           int_for_synchronization_reason(
+               SpecialSynchronizationCases::DeepCopyResourceSynchronization))
+    device_label += " DeepCopyResource)";
+  else
+    device_label += " Instance " + std::to_string(eid.instance_id) + ")";
+
+  return device_label;
+}
+
+bool suppressCounts() {
+  static bool value       = false;
+  static bool initialized = false;
+
+  if (initialized)
+    return value;
+  else {
+    const char* varVal = std::getenv("KOKKOS_PROFILE_SUPPRESS_COUNTS");
+    if (varVal) {
+      std::string v = std::string(varVal);
+      // default to false
+      if (v == "1" || v == "ON" || v == "on" || v == "TRUE" || v == "true" ||
+          v == "YES" || v == "yes")
+        value = true;
+      else
+        value = false;
+    }
+    initialized = true;
+    return value;
+  }
```

Comment:
```suggestion
  static bool value       = [](){
    const char* varVal = std::getenv("KOKKOS_PROFILE_SUPPRESS_COUNTS");
    if (varVal) {
      std::string v = std::string(varVal);
      // default to false
      if (v == "1" || v == "ON" || v == "on" || v == "TRUE" || v == "true" ||
          v == "YES" || v == "yes")
        return true;
    }
    return false;
  }();
  return value;
```
to make it a little more concise.

Reply:
Since `KOKKOS_PROFILE_LIBRARY` is deprecated, I would rather choose `KOKKOS_TOOLS_LOGGER_SUPPRESS_COUNTS` to also make clear that this *is* specific to the kernel logger tool.

Reply:
We can do that as well. Lemme fix.

Reply:
Done.  See below.